webapp/api/_onchain.ts

```diff
@@ -0,0 +1,181 @@
+import { ethers } from "ethers";
+import axios from "axios";
+import ORACLE_ABI from "./abis/oracle.json";
+import wstETH_ABI from "./abis/wstETH.json";
+import rETH_ABI from "./abis/rETH.json";
+import sAVAX_ABI from "./abis/sAVAX.json";
+
+const CHAIN_IDS: {
+  [chainName: string]: number;
+} = {
+  ETH_MAINNET: 1,
+  AVAX_MAINNET: 43114,
+};
+
+const ORACLES: {
+  [chainId: number]: string;
+} = {
+  1: "0x789cD7AB3742e23Ce0952F6Bc3Eb3A73A0E08833",
+  777: "0xb5711dAeC960c9487d95bA327c570a7cCE4982c0",
+  43114: "0x108abfBa5AD61bd61A930BFe73394558d60f0b10",
+};
+
+const ASSETS: {
+  [chainId: number]: { [asset: string]: string };
+} = {
+  1: {
+    USDC: "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
+    WBTC: "0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599",
+    WETH: "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
+    wstETH: "0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0",
+    rETH: "0xae78736Cd615f374D3085123A210448E74Fc6393",
+    AAVE: "0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9",
+    UNI: "0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984",
+    SPELL: "0x090185f2135308BaD17527004364eBcC2D37e5F6",
+    BADGER: "0x3472A5A71965499acd81997a54BBA8D852C6E53d",
+    BAL: "0xba100000625a3754423978a60c9317c58a424e3D",
+    PERP: "0xbC396689893D065F41bc2C6EcbeE5e0085233447",
+  },
+  777: {
+    sAMB: "0x683aae5cD37AC94943D05C19E9109D5876113562",
+  },
+  43114: {
+    WAVAX: "0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7",
+    sAVAX: "0x2b2C81e08f1Af8835a78Bb2A90AE924ACE0eA4bE",
+  },
+};
+
+const PROVIDERS: { [chainId: number]: ethers.providers.StaticJsonRpcProvider } =
+  {
+    1: new ethers.providers.StaticJsonRpcProvider(
+      "https://eth-mainnet.alchemyapi.io/v2/vI8OBZj4Wue9yNPSDVa7Klqt-UeRywrx"
```

Comment:
should use env vars for these

Reply:
Fixed in [65cef25](https://github.com/ribbon-finance/ribbon-frontend/pull/720/commits/65cef25f07217ca23b57a40e76a26a16a0515f3f)